src/julia_internal.h

```diff
@@ -422,7 +422,7 @@ jl_svec_t *jl_perm_symsvec(size_t n, ...);
 
 // this sizeof(__VA_ARGS__) trick can't be computed until C11, but that only matters to Clang in some situations
 #if !defined(__clang_analyzer__) && !(defined(_COMPILER_ASAN_ENABLED_) || defined(_COMPILER_TSAN_ENABLED_))
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
Perhaps also leave a comment as to why this is needed, so people looking at this in the future don't get confused

Reply:
Done!